### ISSUE TITLE
Find incorrect inheritance order

### ISSUE THREAD

**Issue by sobolevn** [PyCQA/flake8-pyi#276]
Right now some modules contain a code like this:

```python
class ResourceLoader(Loader): ...

# ...
# Many-many lines more
# ...

class Loader: ...
```

And since mypy allows this for stubs, it is not marked as an error.
But, today I was really confused finding this in https://github.com/python/typeshed/blob/875f0ca7fcc68e7bd6c9b807cdceeff8a6f734c8/stdlib/importlib/abc.pyi

I was like "What the heck, what's `Loader`? It must be imported somewhere..."

Since it is breaking a very basic python reading order, I propose to forbid such ordering.
`Loader` must be defined before its subclasses.

**Comment by AlexWaygood** [comment]
> Could we simply get rid of the override of `CLASSDEF` since that's what allows the forward declarations to happen in the first place (assuming we're ok with a generic flake error message and not our own)?
> From some quick testing, it seems to work.

That sounds worth trying out! If you file a PR, we can analyse the results of the `typeshed_primer` workflow, and see if it looks like the change causes flake8 to start emitting false positives on typeshed.

In general, less awful monkeypatching of `pyflakes` == good!
